modular_skyrat/modules/modular_items/code/summon_beacon.dm

```diff
@@ -0,0 +1,252 @@
+/obj/item/summon_beacon
+	name = "summoner beacon"
+	desc = "Summons a thing. Probably shouldn't use this one, though."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "gangtool-blue"
+	inhand_icon_state = "radio"
+	w_class = WEIGHT_CLASS_SMALL
+
+	/// How many uses the beacon has left
+	var/uses = 1
+
+	/// A list of allowed areas that the atom can be spawned in
+	var/list/allowed_areas = list(
+		/area,
+	)
+
+	/// A list of possible atoms available to spawn
+	var/list/selectable_atoms = list(
+		/obj/item/summon_beacon,
+	)
+
+	/// The currently selected atom, if any
+	var/atom/selected_atom = null
+
+	/// Descriptor of what area it should work in
+	var/area_string = "any"
+
+	/// If the supply pod should stay or not
+	var/supply_pod_stay = FALSE
+
+/obj/item/summon_beacon/examine()
+	. = ..()
+	. += span_warning("Caution: Only works in [area_string].")
```

Comment:
cant this just be initial(allowed_area or something

Reply:
allowed_area is a list